openmc/tallies.py

```diff
@@ -762,6 +762,8 @@ def _can_merge_filters(self, other):
 
                 # If filters are the second mergeable filters encountered
                 elif filter1.can_merge(filter2) and merge_filters:
+                    merge_filters = True
+                    mergeable_filter = True
```

Comment:
Is there any reason these lines were added? Since they are local variables with a return statement right after they are set, it seems setting has no impact.